libs/vkd3d-shader/spirv.c

```diff
@@ -2843,6 +2944,11 @@ static void vkd3d_dxbc_compiler_emit_dereference_register(struct vkd3d_dxbc_comp
     {
         indexes[index_count++] = vkd3d_dxbc_compiler_emit_register_addressing(compiler, &reg->idx[1]);
     }
+    else if (reg->type == VKD3DSPR_SAMPLER)
+    {
+        /* SM 5.1 will have an index here referring to something which we throw away. */
```

Comment:
When the dereference happens to the sampler, it will always have an index on SM 5.1 it seems, but since we don't emit arrays of resources yet, it's not relevant.